tests/test_gp_interp.py

```diff
@@ -1077,218 +309,66 @@ def test_yaml():
-        rtol = 0.05
-    else:
-        guesses = [0.03, 0.3, 3.0]
-        rtol = 0.05
-    ntrain, nvalidate, nvisualize = 100, 1, 1
-    rng = galsim.BaseDeviate(8675309)
-
-    training_data, validation_data, visualization_data = make_anisotropic_grf_psf_params(
-            ntrain, nvalidate, nvisualize)
-
-    var1s = []
-    var2s = []
-    corrs = []
-
-    for guess in guesses:
-        # noise of 0.3 turns out to be pretty significant here.
-        stars = params_to_stars(training_data, noise=0.03, rng=rng)
-        kernel = "1*AnisotropicRBF(scale_length={0!r})".format([guess, guess])
-        kernel += " + WhiteKernel(1e-5, (1e-7, 1e-1))"
-        interp = piff.GPInterp(kernel=kernel)
-        stars = [mod.fit(s) for s in stars]
-        stars = interp.initialize(stars)
-        interp.solve(stars)
-
-        invLam = interp.gp.kernel_.get_params()['k1__k2__invLam']
-        Lam = np.linalg.inv(invLam)
-        var1s.append(Lam[0, 0])
-        var2s.append(Lam[1, 1])
-        corrs.append(Lam[0, 1] / np.sqrt(Lam[0, 0]*Lam[1, 1]))
-        print(var1s[-1], var2s[-1], corrs[-1])
-    # Check that the inferred correlation is close to the input correlation with params:
-    # var1 = 0.1**2, var2 = 0.2**2, corr = 0.7
-    np.testing.assert_allclose(var1s, 0.1**2, rtol=1.0)  # Only get right order-of-magnitude or so
-    np.testing.assert_allclose(var2s, 0.2**2, rtol=1.0)  # Only get right order-of-magnitude or so
-    np.testing.assert_allclose(corrs, 0.7, rtol=0.1)  # This one works much better
-    # More interesting however, is how independent is the optimization wrt the initial value.
-    # So check that the standard deviation of the results is small.
-    np.testing.assert_array_less(np.std(var1s), 0.1**2*rtol)
-    np.testing.assert_array_less(np.std(var2s), 0.2**2*rtol)
-    np.testing.assert_array_less(np.std(corrs), 0.7*rtol)
-
-
-if __name__ == '__main__':
-    # import cProfile, pstats
-    # pr = cProfile.Profile()
-    # pr.enable()
-    test_constant_psf()
-    test_polynomial_psf()
-    test_grf_psf()
-    test_vonkarman_psf()
-    test_anisotropic_rbf_kernel()
-    test_gp_with_kernels()
-    test_vonkarman_kernel()
-    test_anisotropic_vonkarman_kernel()
+    piff.piffify(config, logger)
+    psf = piff.read(psf_file)
+    assert type(psf.model) is piff.GSObjectModel
+    assert type(psf.interp) is piff.GPInterp
+    print('nstars = ',len(psf.stars))
+    target = psf.stars[17]
+    test_star = psf.interp.interpolate(target)
+    np.testing.assert_almost_equal(test_star.fit.params, target.fit.params, decimal=3)
+    # This should also work if the target doesn't have a fit yet.
+    print('interpolate ',piff.Star(target.data,None))
+    test_star = psf.interp.interpolate(piff.Star(target.data,None))
+    np.testing.assert_almost_equal(test_star.fit.params, target.fit.params, decimal=3)
+
+if __name__ == "__main__":
+
+    test_gp_interp_isotropic()
+    test_gp_interp_anisotropic()
```

Comment:
These two tests are taking over a minute each, which is really too long for the regular unit tests runs.
I think they can be sped up using my usual trick of doing something smaller when run by pytest, but the long thing when run from the command line:
```
if __name__ == "__main__":
    nstars = [160, 160, 400, 400]
else:
    nstars = [1600, 1600, 4000, 4000]
```
Sometimes in conjunction with a larger `rtol`.

I'm not sure if `nstars` is the right thing to adjust here, or if so, how small to make it.  So could you think about whether there is an appropriate adjustment you could make on these tests to speed them up for pytest runs?